src/test/java/CompatibilityTest.java

```diff
@@ -31,6 +33,32 @@ public void v2() throws Exception {
         assertFieldType(i, String[].class, "y");
         assertFieldType(i, Object.class, "z");
     }
+    
+    @Test
+    public void testClassesAreOnlyReWrittenWhenNeeded() throws Exception {
+        System.out.println("rewrtting tests");
+        final Transformer t = new Transformer();
+        AntClassLoader cl = new AntClassLoader() {
+            @Override
+            protected Class<?> defineClassFromData(File container, byte[] classData, String classname) throws IOException {
+                byte[] rewritten = t.transform(classname, classData);
+                if (rewritten!=classData ) {
+                    fail(classname + " was rewritten without need");
+                }
+                return super.defineClassFromData(container, rewritten, classname);
+            }
+        };
+
+        cl.addPathComponent(new File("target/test-classes/v2"));
```

Comment:
From my experience, the IDE issue is only an issue for a multi-module project and even then, it's dead easy to tell your IDE to use the module dir as the CWD.

@jtnord I assume this test runs in your IDE and from the command line without an issue? I don't get why we'd complicate something that's simple and works.